### PR TITLE
feat: add 101 template for zksync-101

### PR DESCRIPTION
**Pull request by itsacoyote**
# What :computer: 
- Update quickstart templates into 101 template
- Updated zkSync to ZKsync across project

# Why :hand:
- easier single template for all of ZKsync 101
- tests for coverage

# Notes :memo:
- will clean up the `quickstart` templates later when docs and zksync-cli have been updated.

### CODE REVIEW

File: templates/101/contracts/4-paymaster/approval/CrownToken.sol [hunk 1]
```diff
@@ -0,0 +1,16 @@
+// SPDX-License-Identifier: Unlicensed
```
Comment (jennyg0) [inline templates/101/contracts/4-paymaster/approval/CrownToken.sol]: should this be MIT

File: templates/101/utils/index.ts [hunk 2]
```diff
@@ -0,0 +1,201 @@
...
+
+// Load env file
+dotenv.config();
+
+export const getProvider = () => {
+  const rpcUrl = hre.network.config.url;
+  if (!rpcUrl)
+    throw `⛔️ RPC URL wasn't found in "${hre.network.name}"! Please add a "url" field to the network config in hardhat.config.ts`;
+
+  // Initialize zkSync Provider
```
Comment (jennyg0) [inline templates/101/utils/index.ts:18]: ZKsync

Comment (jennyg0) [inline templates/101/utils/index.ts]: a few ZKsync updates in this file

File: templates/101/contracts/4-paymaster/approval/CrownToken.sol (hunk 1 above)
Comment (itsacoyote) [reply templates/101/contracts/4-paymaster/approval/CrownToken.sol]: oh good point

File: templates/101/utils/index.ts (hunk 2 above)
Comment (itsacoyote) [reply templates/101/utils/index.ts]: 😬

**Review by jennyg0 — APPROVED** [review]

**Review by jennyg0 — APPROVED** [review]

**Comment by itsacoyote** [comment]
Closing this PR for https://github.com/matter-labs/zksync-contract-templates/pull/22
There are too many merge conflicts for some reason and rather than deal with cleaning that up, I'm just creating a new PR.

**Pull request closed without merging**